File: RedSheet.DbEntities/Models/vProjectImplementationPlan.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectImplementationPlans", Schema = "dbo")]
	[Serializable]
	public partial class vProjectImplementationPlan  
	{
 
	[Required] 
	[MaxLength(1000)]
	public string Activity { get; set; } 
 
	[Column(TypeName = "date")]
	public Nullable<System.DateTime> EndDate { get; set; } 
 
	[Required]
	public bool IsEvent { get; set; } 
 
	[Required] 
	[MaxLength(62)]
	public string Name { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectImplementationPlanId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 
 
	[Required] 
	[Column(TypeName = "date")]
	public System.DateTime StartDate { get; set; } 

	public vProjectImplementationPlan()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-project-implementation-plan.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectImplementationPlan {
    constructor(vProjectImplementationPlan?: vProjectImplementationPlan )  {
        let properties = [ "activity", "endDate", "isEvent", "name", "projectImplementationPlanId", "projectModuleId", "startDate",];
        for (let property of properties)
            if (vProjectImplementationPlan && vProjectImplementationPlan[property])
                this[property] = vProjectImplementationPlan[property];
    }
 
    @required()
    @maxLength(1000)
	activity : string =   undefined;
 
	endDate : Date =   undefined;
 
	isEvent : boolean = false ;
 
    @required()
    @maxLength(62)
	name : string =   undefined;
 
	projectImplementationPlanId : number =   0 ;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
    @required()
	startDate : Date =   undefined;


}
